subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/ivyresolve/strategy/DefaultVersionComparator.java

```diff
@@ -24,7 +25,18 @@
  * @see StaticVersionComparator
  */
 public class DefaultVersionComparator implements VersionComparator {
-    private final Comparator<Version> baseComparator = new StaticVersionComparator();
+    private Comparator<Version> baseComparator;
+    private final FeaturePreviews featurePreviews;
+
+    public DefaultVersionComparator() {
+        // We preserve the no-arg constructor since this change is temporary
+        // and there is a chance this class is used in plugins out there
+        this(null);
```

Comment:
:x: Add `@Deprecated`?

Reply:
Oh no, this is the constructor that will remain in place. When the feature is integrated in Gradle 7.0, the constructor taking `FeaturePreviews` disappears again.

Reply:
oh, right